src/main/java/org/eugene/controller/Renderer.java

```diff
@@ -1,151 +1,186 @@
-        String directory = "";
-        if(matcher.find()){
-            directory = matcher.group(1);
-        }
-        return directory;
-    }
-
-    public List<List<String>> getData(){
-        return VirtualDB.getInstance().getCommonData().getData();
-    }
-
-    public void refreshTable(){
-        refreshTable(showingList);
-    }
-
-    public void refreshTable(List<String> showingList){
-        CommonData commonData = VirtualDB.getInstance().getCommonData();
-        TableMeta tableMeta = VirtualDB.getInstance().getTableMeta();
-        tableRenderer.refresh(showingList, new ArrayList<String>(commonData.getColumnToType().keySet()), tableMeta.getRow(), tableMeta.getColumn(), commonData.getData());
-    }
-
-    public void refreshAggregationPane(String columnName){
-        CommonData commonData = VirtualDB.getInstance().getCommonData();
-        List<String> typeList = new ArrayList<String>();
-        typeList.add("INTEGER");
-        typeList.add("INT");
-        typeList.add("LONG");
-        typeList.add("DOUBLE");
-        typeList.add("FLOAT");
-        typeList.add("UNION");
-        if(!typeList.contains(commonData.getColumnToType().get(columnName).toUpperCase())){
-            return;
-        };
-        Map<String, String> keyToValue = PhysicalDB.getInstance().getAggregation(columnName);
-        dashboardRenderer.refreshAggregationPane(columnName, keyToValue);
-    }
-
-    public void refreshProportionPane(String columnName){
-        CommonData commonData = VirtualDB.getInstance().getCommonData();
-        List<String> typeList = new ArrayList<String>();
-        typeList.add("INTEGER");
-        typeList.add("INT");
-        typeList.add("LONG");
-        typeList.add("DOUBLE");
-        typeList.add("FLOAT");
-        typeList.add("UNION");
-        if(typeList.contains(commonData.getColumnToType().get(columnName).toUpperCase())){
-            return;
-        }
-        Map<String, Integer> itemToCount = PhysicalDB.getInstance().getProportion(columnName);
-        dashboardRenderer.refreshProportionPane(columnName, itemToCount);
-    }
-
-}
+package org.eugene.controller;
+
+import javafx.stage.FileChooser;
+import javafx.stage.Stage;
+
+import org.apache.commons.lang3.SystemUtils;
+import org.apache.hadoop.fs.Path;
+import org.eugene.core.common.AWSS3Reader;
+import org.eugene.model.CommonData;
+import org.eugene.model.TableMeta;
+import org.eugene.persistent.PhysicalDB;
+import org.eugene.persistent.VirtualDB;
+import org.eugene.ui.Constants;
+import org.eugene.ui.Dashboard;
+import org.eugene.ui.Main;
+import org.eugene.ui.Table;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class Renderer {
+
+	private Stage stage;

```

Comment:
Please set the indent to 4 spaces, instead of 8 spaces.

Reply:
apologies - im still an eclipse person where the defaults are in tabs. 
il change this to 4 spaces and commit

Reply:
Thanks, but seems the tab is still 8 spaces. The correct indent should look like this: 
![image](https://github.com/Eugene-Mark/bigdata-file-viewer/assets/51333308/6a80d9fc-1cef-47f8-9a86-ba7cf3139282)

Reply:
Apologies - I have now formatted it using https://github.com/google/google-java-format
Here is how it looks (before vs after )

![image](https://github.com/Eugene-Mark/bigdata-file-viewer/assets/30702253/93bbb69e-4b3c-4da0-8b19-dcdb32b05b5e)

I will commit this now.
Thanks again!

Reply:
Hi @sadaaithal , thanks for your effort to introduce google java format. However, the project now is not following it, can you simply just set your editor to make tab equals 4 spaces. As for eclipse, you can refer to [this link](https://stackoverflow.com/questions/407929/how-do-i-change-eclipse-to-use-spaces-instead-of-tabs)  about how to do that.

Reply:
hi @Eugene-Mark  ive made the change and it finally reflects the 4 spaced tab. 
Please check and hopefully this works.
Apologies for the back and forth  as im new to github 

best
-s